sympy/polys/tests/test_hypothesis.py

```diff
@@ -4,40 +4,33 @@
 from sympy.polys.polytools import Poly
 
 
-@given(
-    coefficients1=st.lists(st.integers()),
-    coefficients2=st.lists(st.integers()),
-    coefficients3=st.lists(st.integers()),
-)
-def test_gcd_hypothesis(coefficients1, coefficients2, coefficients3):
-    f = Poly(coefficients1, x, domain="ZZ")
-    g = Poly(coefficients2, x, domain="ZZ")
-    r = Poly(coefficients3, x, domain="ZZ")
+def polys(*, nonzero=False, domain="ZZ"):
+    # This is a simple strategy, but sufficient the tests below
+    elems = {"ZZ": st.integers(), "QQ": st.fractions()}
+    coeff_st = st.lists(elems[domain])
```

Comment:
Generating elements of different domains than just ZZ and QQ would be likely to pick up on many more errors. Already ZZ and QQ are the most well used and tested so the least likely to have bugs.